### PR TITLE
Keep current serve after refresh

### LINKED ISSUE #75: Keep the current SERVE/commands after refresh
When the user `refresh` the page, the client won't "remember" the current running `ng serve` command
We should find a way for the client to keep track of the current running commands after a refresh.

### PR DESCRIPTION
**Pull request by ZimGil**
## Keep current `ng serve` command after refresh/close

- Saving and loading from `localStorage` [resolves #75]
  > Saving `ng serve` command ID to `localStorage` upon completed compilation
  > Loading saved command ID `OnInit`
  > Removing command ID from `localStorage` upon sending a `stop command`

- Error for miscalled `stop command`
  > In some cases (server offline / been restarted) a `stop command` would fail
  > Will pop an error and drop the currently held / saved command ID

### CODE REVIEW

File: src/app/app.component.ts [hunk 1]
```diff
@@ -138,6 +140,10 @@ export class AppComponent implements OnInit {
         });
   }
 
+  loadCurrentServe(): void {
```
Comment (ortichon) [inline src/app/app.component.ts]: rename to something like `getServeStatusFromStorage`

File: src/app/app.component.ts [hunk 2]
```diff
@@ -138,6 +140,10 @@ export class AppComponent implements OnInit {
...
+    this.serveCommandId = localStorage.getItem('ngServeCommandId');
```
Comment (ortichon) [inline src/app/app.component.ts]: now when you have the `serveCommandId` query the server for this id status

File: src/app/app.component.ts [hunk 3]
```diff
@@ -34,6 +34,7 @@ export class AppComponent implements OnInit {
   ngOnInit() {
     this.keepAlive();
     this.checkAngularProject();
+    this.loadCurrentServe();
```
Comment (ortichon) [inline src/app/app.component.ts]: flow:
1. `keepAlive`
2. `loadCurrentServe`:
      true -> go to serve screen.
      false -> `checkAngularProject`

File: src/app/components/serve/serve.component.ts [hunk 4]
```diff
@@ -19,17 +20,26 @@ export class ServeComponent {
...
 
   startServing(): void {
     this.sendCommand.emit(this.options.createCommandString());
   }
 
   stopServing(): void {
     this.commandService.stopServing(this.serveCommandId)
-    .subscribe(() => {
-      this.serveCommandId = null;
+    .subscribe(() => {},
```
Comment (ortichon) [inline src/app/components/serve/serve.component.ts]: ugly indentation

File: src/app/components/serve/serve.component.ts [hunk 5]
```diff
@@ -19,17 +20,26 @@ export class ServeComponent {
...
+    error => {
```
Comment (ortichon) [inline src/app/components/serve/serve.component.ts]: same

File: src/app/components/serve/serve.component.ts [hunk 6]
```diff
@@ -19,17 +20,26 @@ export class ServeComponent {
...
+      this.errorService.addError({
+        errorText: 'The "ng serve" command you\'re trying to stop was not found',
+        errorDescription: 'The server is offline or have been restarted since you\'ve run this command'
+      });
     });
+    this.serveCommandId = null;
```
Comment (ortichon) [inline src/app/components/serve/serve.component.ts]: should only be null if the stop command succeeds or fails, so try to add `finally`

File: src/app/components/serve/serve.component.ts [hunk 7]
```diff
@@ -19,17 +20,26 @@ export class ServeComponent {
...
+    localStorage.removeItem('ngServeCommandId');
```
Comment (ortichon) [inline src/app/components/serve/serve.component.ts]: same

**Comment by ZimGil** [comment]
Few Changes were made:

- `localStorage` save time
  > Now will be save immediately when getting the command ID of `ng serve` to make sure the client have it even if refreshed right after sending the command

- The server will wait for running command
  > In case stop command was sent before the server was done compiling, it will now wait until the command is done and then kill it

- `Stop Serving` button is now `disabled` after user click to avoid multiple calls to the server

File: server/server.ts [hunk 8]
```diff
@@ -10,6 +10,7 @@ import { AngularCliProcessStatus } from './models/angular-cli-process-status.enu
 import { AngularProjectChecker } from './angular-project-checker';
 import { printLogo } from './logo-printer.helper';
 import { NgWizLogger } from './ngWizLogger';
+import { timer } from 'rxjs';
```
Comment (ortichon) [inline server/server.ts]: move to correct location

File: src/app/app.component.ts [hunk 9]
```diff
@@ -138,13 +138,35 @@ export class AppComponent implements OnInit {
         });
   }
 
+  checkIfRunningServeCommand(): void {
+    const savedCommandId = localStorage.getItem('ngServeCommandId');
+    if (savedCommandId) {
+      this.commandService.checkCommandStatus(savedCommandId)
+      .subscribe(status => {
```
Comment (ortichon) [inline src/app/app.component.ts]: indentation

File: src/app/components/serve/serve.component.ts [hunk 10]
```diff
@@ -18,18 +19,34 @@ export class ServeComponent {
   @Input() serveCommandId: string;
   command: AngularCliCommand;
   options = new NgserveOptions();
+  isStoppingServeCommand = false;
 
-  constructor(private commandService: CommandService) {}
+  constructor(
+    private commandService: CommandService,
+    private errorService: ErrorService
+    ) {}
```
Comment (ortichon) [inline src/app/components/serve/serve.component.ts]: indentation

**Pull request merged**